lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -37,7 +37,7 @@ class ProxyGenerator
      * Used to match very simple id methods that don't need
      * to be decorated since the identifier is known.
      */
-    const PATTERN_MATCH_ID_METHOD = '((public\s+)?(function\s+%s\s*\(\)\s*)\s*{\s*return\s*\$this->%s;\s*})i';
+    const PATTERN_MATCH_ID_METHOD = '((public\s+)?(function\s+%s\s*\(\)\s*)\s*(?::\s*[\w\\\\]+\s*)?{\s*return\s*\$this->%s;\s*})i';
```

Comment:
@janlanger well, general purpose lib => general purpose solution :-)

In think tests for this particular scenario are somewhere in https://github.com/doctrine/common/blob/faf7d81273eeffb84d524006daf85d37a14ac42a/tests/Doctrine/Tests/Common/Proxy/ProxyClassGeneratorTest.php